packages/gestalt/src/Masonry/fullWidthMultiColumLayout.js

```diff
@@ -0,0 +1,52 @@
+// @flow strict
+import { type Cache } from './Cache';
+import multiColumnLayout from './multiColumnLayout';
+import { type Position } from './types';
+
+const fullWidthLayout = <T>({
+  gutter = 0,
+  minCols = 2,
+  idealColumnWidth = 240,
+  width,
+  measurementCache,
+  ...otherProps
+}: {
+  gutter?: number,
+  minCols?: number,
+  idealColumnWidth?: number,
+  width?: ?number,
+  logWhitespace?: (number) => void,
+  measurementCache: Cache<T, number>,
+  positionCache: Cache<T, Position>,
+  whitespaceThreshold?: number,
+}): ((items: $ReadOnlyArray<T>) => $ReadOnlyArray<Position>) => {
+  if (width == null) {
+    return (items) =>
+      items.map(() => ({
+        top: Infinity,
+        left: Infinity,
+        width: Infinity,
+        height: Infinity,
+      }));
+  }
+  // "This is kind of crazy!" - you
+  // Yes, indeed. The "guessing" here is meant to replicate the pass that the
+  // original implementation takes with CSS.
+  const colguess = Math.floor(width / idealColumnWidth);
+  const columnCount = Math.max(Math.floor((width - colguess * gutter) / idealColumnWidth), minCols);
+  const columnWidth = Math.floor(width / columnCount) - gutter;
+
+  return (items: $ReadOnlyArray<T>) =>
```

Comment:
instead of a forked layout, it seems like we could move this into `fullWidthLayout` as a conditional path. what do you thinka bout doing that? i think the ultimate goal was always to have the `multiColumnLayout` logic live inside the "base" layouts so awesome to see this